examples/text-classification/new_tf_text_classification.py

```diff
@@ -0,0 +1,498 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Fine-tuning the library models for sequence classification."""
+# You can also adapt this script on your own text classification task. Pointers for this are left as comments.
+
+import logging
+import os
+import random
+import sys
+from dataclasses import dataclass, field
+from typing import Optional
+from math import ceil
+from pathlib import Path
+
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '1'  # Reduce the amount of console output from TF
+
+import numpy as np
+import tensorflow as tf
+from datasets import load_dataset, load_metric
+
+import transformers
+from transformers import (
+    AutoConfig,
+    TFAutoModelForSequenceClassification,
+    AutoTokenizer,
+    DataCollatorWithPadding,
+    EvalPrediction,
+    HfArgumentParser,
+    PretrainedConfig,
+    TrainingArguments,
+    default_data_collator,
+    set_seed,
+)
+from transformers.trainer_utils import is_main_process, get_last_checkpoint
+from transformers.utils import check_min_version
+
+logger = logging.getLogger(__name__)
+
+# region Helper classes
+class DataSequence(tf.keras.utils.Sequence):
```

Comment:
Maybe too complex for this example, but I believe it's generally best to have the longest sequence be part of the initial batch, as the full batch then has the maximum (`seq_len x batch_size`) that will be encountered during training; if the training was to OOM at some point, it would do so right off the bat rather than 1hr in.

Unless I'm mistaken, that's not done here, would it be easy to add?